CONTRIBUTING.md

```diff
@@ -82,3 +81,29 @@ Before a pull request can be merged:
 * The feature branch must be rebased so it contains the latest changes from the target branch
 * The CI has to pass successfully
 * Every comment has to be addressed and resolved
+
+## Testing
+
+Tests are written using the *[Ginkgo](https://onsi.github.io/ginkgo/)* framework. Here are some general advices when writing tests:
+
+* When the global `Describe` doesn't add enough context to the tests, use a nested `Describe` or a `Context` to specify what this test alludes to. Contexts should always start with _When_ (ie. "_When calling the function Foo_")
+* Start the descriptions of `It` blocks in lowercase and try to be as descriptive as possible
+* Avoid ignoring errors. In other words, make sure all of them are caught and tested
+* Files ending with `_suite_test.go` are meant to store the code that is common for tests in the same directory and that will be executed before them. Use this files only if your test setup is big enough to justify it (ie. the suite file has more than the test suite name and the teardown)
+* When required, remember to add the `CRD`'s at the `envtest` setup, for instance: [release-service/release_suite_test.go at main · redhat-appstudio/release-service · GitHub](https://github.com/redhat-appstudio/release-service/blob/main/controllers/release/release_suite_test.go#L65) - remembering this saves a lot of time
```

Comment:
add the CRDs during the envtest setup? in the envtest setup? at feels weird to me